### DISCUSSION TITLE
`cargo 3ds` commands take very very long

### DISCUSSION THREAD

**Discussion by CenTdemeern1** [rust3ds/ctru-rs#201] [Q&A] [answered]
Something like `cargo 3ds check` takes 15 seconds while for an equivalent program it takes `cargo check` a negligible amount of time (0.02s).
During a lot of this time there's no output, so I can't really tell what's going on. The majority of the wait seems to happen before it even tries to start compiling.
Is this an unfortunate consequence of cross-compilation or is there something else going on here? (I'm on an M2 Mac Mini, if that matters)

**Comment by Meziu** [comment]
This is not normal behaviour. `cargo-3ds` does little more than calling `cargo check` itself when used that way, so it shouldn't add any more than a negligible wait to the execution of the command. Please share logs of your execution (with `--verbose`) and offer more context about the project that made you notice this issue.

**Reply by CenTdemeern1** [reply]
Here is a recording of my terminal session: https://asciinema.org/a/wCeQ0bgqmPlfUQtEB9OMH4i56
`--verbose` didn't seem to really show much during the downtime.

**Reply by Meziu** [reply]
I am unable to replicate the issue with the latest `cargo-3ds` on the `ctru-rs` project folder. Is your issue present for this repo? Does it occur if you use this command instead?

```bash
cargo check --target armv6k-nintendo-3ds -Zbuild-std
```

There might be a particular compilation slowdown when `cargo-check`ing for the 3ds' target triple, maybe in one of your dependencies.

**Reply by ian-h-chamberlain** [reply]
You can also try `cargo 3ds --verbose check`, which will display more info about what `cargo-3ds` itself is doing (as opposed to `check --verbose` which just passes `--verbose` down to Cargo).

**Reply by CenTdemeern1** [reply]
> I am unable to replicate the issue with the latest `cargo-3ds` on the `ctru-rs` project folder. Is your issue present for this repo? Does it occur if you use this command instead?
> 
> ```shell
> cargo check --target armv6k-nintendo-3ds -Zbuild-std
> ```
> 
> There might be a particular compilation slowdown when `cargo-check`ing for the 3ds' target triple, maybe in one of your dependencies.

Yes: https://asciinema.org/a/swDikLK4ZJbi9jf2txsCYnZAH

> You can also try `cargo 3ds --verbose check`, which will display more info about what `cargo-3ds` itself is doing (as opposed to `check --verbose` which just passes `--verbose` down to Cargo).

It seems to hang when running the second command it runs: https://asciinema.org/a/WzJo2rl94EGc8GSodWx0LYQMM

**Comment by CenTdemeern1** [comment, marked as answer]
I'm pretty confident I found the problem: #233
